From 221c6115e3f8cccc4dc9756c45bf22734e6f7d64 Mon Sep 17 00:00:00 2001
From: Radu Bulie <radu.bulie@freescale.com>
Date: Wed, 17 Jul 2013 09:26:59 +0000
Subject: [PATCH 064/130] dpa_offload: Add direct scheme support for
 classifier miss action

This patch enables the direct schemesupport  when a miss action is configured.

Signed-off-by: Radu Bulie <radu.bulie@freescale.com>
Change-Id: Ie16ab1e611a06b026fb5927ec2efea314bb0e2e1
Reviewed-on: http://git.am.freescale.net:8181/3354
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Chereji Marian-Cornel-R27762 <marian.chereji@freescale.com>
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
Tested-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
[Fetch from http://git.freescale.com/git/cgit.cgi/ppc/sdk/linux.git/,
 Tag: fsl-sdk-v1.4.5
 rebase on current context.]
Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 drivers/staging/fsl_dpa_offload/dpa_classifier.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/staging/fsl_dpa_offload/dpa_classifier.c b/drivers/staging/fsl_dpa_offload/dpa_classifier.c
index 34758c7..a39fc69 100644
--- a/drivers/staging/fsl_dpa_offload/dpa_classifier.c
+++ b/drivers/staging/fsl_dpa_offload/dpa_classifier.c
@@ -430,7 +430,8 @@ int dpa_classif_table_modify_miss_action(int			td,
 
 	/* Fill the [miss_engine_params] structure w/ data */
 	errno = action_to_next_engine_params(miss_action, &miss_engine_params,
-		NULL, NULL, NULL);
+					NULL, ptable->params.distribution,
+					ptable->params.classification);
 	if (errno < 0) {
 		RELEASE_OBJECT(ptable);
 		log_err("Failed verification of miss action params for table "
-- 
1.7.5.4

